### PR TITLE
Decode base64 auth header

### LINKED ISSUE #895: DockerPushImage using invalid format for X-Registry-Auth
### Expected Behavior
If I have a docker config file that works with the docker cli, the same config could be consumed successfully by the plugin. The `X-Registry-Auth` header would match the documented format.  It seems the auth field in the `~/docker/config.json` should be decoded and re-encoded in the documented format with the explicit user and password fields.

### Current Behavior
Currently it seems the auth field is used directly and encoded in the `X-Registry-Auth` header but fails to authorize against the registry.

It seems like in 6.0.0, the plugin no longer _always_ uses the supplied registryCredentials, and instead will try to use the config file if it exists.  Seems like it would be preferable to have this behavior optional.

### Context
I am trying to push to a remote registry via the DockerPushImage task.

### Steps to Reproduce (for bugs)
I am using version 6.0.0.  When I have no `~/.docker/config.json` file, the credentials I supply in the `registryCredentials` configuration block are properly read and the following header is included in the POST to push

```
2019-12-08T13:45:24.388-0800 [DEBUG] [com.bmuschko.gradle.docker.shaded.org.apache.http.wire] http-outgoing-41 >> "POST /images/my.docker.registry/test/image:latest/push HTTP/1.1[\r][\n]"
2019-12-08T13:45:24.388-0800 [DEBUG] [com.bmuschko.gradle.docker.shaded.org.apache.http.wire] http-outgoing-41 >> "Accept: application/json[\r][\n]"
2019-12-08T13:45:24.388-0800 [DEBUG] [com.bmuschko.gradle.docker.shaded.org.apache.http.wire] http-outgoing-41 >> "User-Agent: Jersey/2.27 (Apache HttpClient 4.5.6)[\r][\n]"
2019-12-08T13:45:24.388-0800 [DEBUG] [com.bmuschko.gradle.docker.shaded.org.apache.http.wire] http-outgoing-41 >> "X-Registry-Auth: eyJ1c2VybmFtZSI6Im15dXNlciIsInBhc3N3b3JkIjoibXlwYXNzd29yZCIsInNlcnZlcmFkZHJlc3MiOiJodHRwczovL215LmRvY2tlci5yZWdpc3RyeSJ9[\r][\n]"
```
The base64 content decodes as follows :

```
{"username":"myuser","password":"mypassword","serveraddress":"https://my.docker.registry"}
```
This matches the format documented here : https://docs.docker.com/engine/api/v1.39/#section/Authentication

If I have a `~/.docker/config.json` the `auth` encoded field is used directly and the POST to push fails.

My `~/.docker/config.json` is as follows :
```
{
  "auths" : {
      "https://my.docker.registry" : {
         "auth": "bXl1c2VyOm15cGFzc3dvcmQ="
      }
  }
}
```
The push call with the docker cli works as expected.

But the following header is used in the POST via the Gradle plugin :
```
2019-12-08T13:44:41.713-0800 [DEBUG] [com.bmuschko.gradle.docker.shaded.org.apache.http.wire] http-outgoing-39 >> "POST /images/my.docker.registry/test/image:latest/push HTTP/1.1[\r][\n]"
2019-12-08T13:44:41.713-0800 [DEBUG] [com.bmuschko.gradle.docker.shaded.org.apache.http.wire] http-outgoing-39 >> "Accept: application/json[\r][\n]"
2019-12-08T13:44:41.713-0800 [DEBUG] [com.bmuschko.gradle.docker.shaded.org.apache.http.wire] http-outgoing-39 >> "User-Agent: Jersey/2.27 (Apache HttpClient 4.5.6)[\r][\n]"
2019-12-08T13:44:41.713-0800 [DEBUG] [com.bmuschko.gradle.docker.shaded.org.apache.http.wire] http-outgoing-39 >> "X-Registry-Auth: eyJzZXJ2ZXJhZGRyZXNzIjoiaHR0cHM6Ly9teS5kb2NrZXIucmVnaXN0cnkiLCJhdXRoIjoiYlhsMWMyVnlPbTE1Y0dGemMzZHZjbVE9In0=[\r][\n]"
```

This decodes as follows :
```
{"serveraddress":"https://my.docker.registry","auth":"bXl1c2VyOm15cGFzc3dvcmQ="}
```

### Your Environment
Gradle 6.0.1
Plugin 6.0.0
docker version
```
Client: Docker Engine - Community
 Version:           19.03.5
 API version:       1.40
 Go version:        go1.12.12
 Git commit:        633a0ea
 Built:             Wed Nov 13 07:22:34 2019
 OS/Arch:           darwin/amd64
 Experimental:      false

Server: Docker Engine - Community
 Engine:
  Version:          19.03.5
  API version:      1.40 (minimum version 1.12)
  Go version:       go1.12.12
  Git commit:       633a0ea
  Built:            Wed Nov 13 07:29:19 2019
  OS/Arch:          linux/amd64
  Experimental:     true
 containerd:
  Version:          v1.2.10
  GitCommit:        b34a5c8af56e510852c35414db4c1f4fa6172339
 runc:
  Version:          1.0.0-rc8+dev
  GitCommit:        3e425f80a8c931f88e6d94a8c831b9d5aa481657
 docker-init:
  Version:          0.18.0
  GitCommit:        fec3683
```

### PR DESCRIPTION
**Pull request by kgrech**
closes #895 

The PR adds a required decode method call applied to the content of the auth section of the config.json file plus extra functional tests.

I had to update gradle to v 5.6.4 as there was a need in ``withEnvironment`` method of the ``GradleRunner`` class added in 5.2.0 only

### CODE REVIEW

**Comment by kgrech** [comment]
@bmuschko Any idea on why is ``DockerWaitHealthyContainerFunctionalTest`` failing? It's passing on my local... Maybe it is related to gradle version update?

**Comment by kgrech** [comment]
@bmuschko  See https://github.com/bmuschko/gradle-docker-plugin/pull/903 would you mind me disabling ``DockerWaitHealthyContainerFunctionalTest > Invoke onNext periodically passing the health status``?

**Comment by bmuschko** [comment]
@kgrech I don't think the test failure it's related to your change. It's a flaky test that needs fixing.

File: gradle/wrapper/gradle-wrapper.properties [hunk 1]
```diff
@@ -1,5 +1,5 @@
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-5.1.1-bin.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-5.6.4-bin.zip
```
Comment (bmuschko) [inline gradle/wrapper/gradle-wrapper.properties]: Can you refer this change? We want to maintain Gradle 5.1 compatibility. Switching to 5.6 will not make it backward compatible anymore.

Comment (kgrech) [reply gradle/wrapper/gradle-wrapper.properties]: See in the PR description. I need a withEnv method on GradleRunner. Do you know alternative way?

Comment (bmuschko) [reply gradle/wrapper/gradle-wrapper.properties]: Got it. Let's go with Gradle 5.2 then. That's the [version that introduced the method](https://docs.gradle.org/current/javadoc/org/gradle/testkit/runner/GradleRunner.html#withEnvironment-java.util.Map-). Can you please also update the [minimum compatible Gradle version in the user guide](https://github.com/bmuschko/gradle-docker-plugin/blob/master/src/docs/asciidoc/00-intro.adoc)?

Comment (kgrech) [reply gradle/wrapper/gradle-wrapper.properties]: Sure, will do so!

Comment (kgrech) [reply gradle/wrapper/gradle-wrapper.properties]: @bmuschko done, have a look

**Comment by bmuschko** [comment]
Thanks for your work!

**Pull request merged**